features/lessons/src/main/java/com/cryart/sabbathschool/lessons/ui/readings/SSReadingActivity.kt

```diff
@@ -0,0 +1,189 @@
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.cryart.sabbathschool.lessons.ui.readings
+
+import android.graphics.Color
+import android.graphics.Typeface
+import android.graphics.drawable.BitmapDrawable
+import android.os.Bundle
+import android.view.Menu
+import android.view.MenuItem
+import androidx.core.view.ViewCompat
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.viewpager.widget.ViewPager
+import coil.load
+import com.cryart.sabbathschool.core.extensions.prefs.SSPrefs
+import com.cryart.sabbathschool.core.misc.SSColorTheme
+import com.cryart.sabbathschool.core.misc.SSConstants
+import com.cryart.sabbathschool.core.misc.SSUnzip
+import com.cryart.sabbathschool.lessons.R
+import com.cryart.sabbathschool.lessons.data.model.SSLessonInfo
+import com.cryart.sabbathschool.lessons.data.model.SSRead
+import com.cryart.sabbathschool.lessons.data.model.SSReadComments
+import com.cryart.sabbathschool.lessons.data.model.SSReadHighlights
+import com.cryart.sabbathschool.lessons.databinding.SsReadingActivityBinding
+import com.cryart.sabbathschool.lessons.ui.base.SSBaseActivity
+import com.google.firebase.storage.FirebaseStorage
+import com.google.firebase.storage.StorageMetadata
+import com.google.firebase.storage.StorageReference
+import com.mikepenz.google_material_typeface_library.GoogleMaterial
+import com.mikepenz.iconics.IconicsDrawable
+import java.io.File
+import timber.log.Timber
+
+class SSReadingActivity : SSBaseActivity(), SSReadingViewModel.DataListener, ViewPager.OnPageChangeListener {
+    private val binding: SsReadingActivityBinding by lazy { SsReadingActivityBinding.inflate(layoutInflater) }
+    private val latestReaderArtifactRef: StorageReference = FirebaseStorage.getInstance().reference.child(SSConstants.SS_READER_ARTIFACT_NAME)
+    private lateinit var ssReadingViewModel: SSReadingViewModel
+    private lateinit var prefs: SSPrefs
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(binding.root)
+
+        prefs = SSPrefs(this)
```

Comment:
We can inject this dependency with dagger.
The activity will need an @AndroidEntryPoint annotation

Reply:
Ideally it should be moved to viewmodel, can do this later after modularization